go/vt/vtgate/planbuilder/route.go

```diff
@@ -204,6 +225,23 @@ func (rb *route) Wireup(plan logicalPlan, jt *jointab) error {
 	return nil
 }
 
+// prepareTheAST does minor fixups of the SELECT struct before producing the query string
+func (rb *route) prepareTheAST() {
+	_ = sqlparser.Walk(func(node sqlparser.SQLNode) (bool, error) {
+		switch node := node.(type) {
+		case *sqlparser.Select:
+			if len(node.SelectExprs) == 0 {
+				node.SelectExprs = []sqlparser.SelectExpr{
+					&sqlparser.AliasedExpr{
+						Expr: sqlparser.NewIntLiteral([]byte{'1'}),
+					},
+				}
+			}
+		}
```

Comment:
sometimes we don't need anything from a route except the number of matching rows. in those cases, we add a single literal because a SELECT with no expressions is not valid